lib/Service.js

```diff
@@ -131,8 +131,10 @@ class Service extends Component {
         delete this._pendingRequests[id];
 
         // ** Check for errors
-        if (res.error)
-            return req.fail(res.error);
+        if (res.error){
+            var err = errors.isError(res.error) ? res.error.toObject() : res.error;
```

Comment:
great observation @bradserbu I saw the isError inside the errors class and assumed it was a custom function :blush: